### PR TITLE
Add more guild fields

### PR DESCRIPTION
**Pull request by AregevDev**
Good day,
This pull request adds five new fields to the `Guild` struct and `PartialGuild` struct
* premium_tier
* premium_subscription_count
* description
* banner
* vanity_url_code

For each of these, I updated the test resources and made sure `cargo test --all-features` pass correctly
In additon I have created a dummy bot and brought him to our server to test out the new fields, it all works as expected

### Notice during the review
Unused import at `src/model/guild/mod.rs:37`
JSON resources

Have a nice day
AregevDev

### CODE REVIEW

File: src/model/guild/premium_tier.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    pub fn num(self) -> u64 {
+        match self {
+            PremiumTier::Tier0 => 0,
+            PremiumTier::Tier1 => 1,
+            PremiumTier::Tier2 => 2,
+            PremiumTier::Tier3 => 3,
+            PremiumTier::__Nonexhaustive => unreachable!(),
+        }
+    }
+}
```
Comment (arqunis) [inline src/model/guild/premium_tier.rs]: nitpick: Add a trailing newline.
```suggestion
}

```

**Review by arqunis — CHANGES_REQUESTED** [review]

**Comment by AregevDev** [comment]
Haha, didn't pay attention :))

File: src/model/guild/mod.rs [hunk 2]
```diff
@@ -32,6 +34,8 @@ use parking_lot::RwLock;
 #[cfg(all(feature = "cache", feature = "model"))]
 use std::sync::Arc;
 #[cfg(feature = "model")]
+use crate::http;
```
Comment (arqunis) [inline src/model/guild/mod.rs]: https://github.com/serenity-rs/serenity/pull/600#issue-286393882: `Unused import at src/model/guild/mod.rs:37`
Is this import required then if the compiler states that it's unused?
I don't see any usage of the http module in your code, so I'm wondering why you've added it.

**Review by arqunis — CHANGES_REQUESTED** [review]

File: src/model/guild/mod.rs (hunk 2 above)
Comment (AregevDev) [reply src/model/guild/mod.rs]: I didn't, did I?
Perhaps I checked out something incorrectly...
My bad

**Review by arqunis — APPROVED** [review]

**Pull request merged**